### PR TITLE
feat: add catalog UI to browse the catalog

### PR DESCRIPTION
**Pull request by benoitf**
related to https://github.com/containers/podman-desktop/issues/8972

https://github.com/user-attachments/assets/597060a1-d189-45c7-9a01-d1d7595df7b8

Change-Id: I820ba559ae561308bca38dc867439615d6e8d8af

### CODE REVIEW

**Comment by benoitf** [comment]
> I am not sure to follow, is this going to be iframed in the website ?

yes, to reuse most of UI svelte components

**Comment by benoitf** [comment]
how it renders within the website:

https://github.com/user-attachments/assets/ac0d6104-523b-4993-a4d7-15420eb0cf23

File: playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  webServer: {
+    command: 'npm run build && npm run preview',
+    port: 4173,
+  },
+  testDir: 'tests',
+  // eslint-disable-next-line sonarjs/slow-regex
+  testMatch: /(.+\.)?(test|spec)\.[jt]s/,
+};
+
+export default config;
```
Comment (axel7083) [inline playwright.config.ts:13]: we do not have playwright test in this ?

File: tsconfig.node.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
+  "include": ["*.config.ts", "*.config.js", "*.config.mjs", "eslint.config.mjs"]
+}
```
Comment (axel7083) [inline tsconfig.node.json:4]: node tsconfig ? for a static website ? is this for playwright ?

**Review by axel7083 — COMMENTED** [review]
Why using `static/api` folder structure ? static do not really provide apis... ?

File: playwright.config.ts (hunk 1 above)
Comment (benoitf) [reply playwright.config.ts:13]: I'll remove the file generated by sveltekit

File: tsconfig.node.json (hunk 2 above)
Comment (benoitf) [reply tsconfig.node.json:4]: it's for the linter

**Comment by benoitf** [comment]
> Why using static/api folder structure ? static do not really provide apis... ?

it's to have the current content of the website https://github.com/containers/podman-desktop-catalog/tree/gh-pages/api at the proper location

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.DS_Store
+node_modules
+/dist
+/.svelte-kit
+/.eslintcache
+/coverage
+/package
```
Comment (jeffmaury) [inline .gitignore:7]: ?

File: src/app.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<meta charset="utf-8" />
+		<link rel="icon" href="%sveltekit.assets%/favicon.png" />
+		<meta name="viewport" content="width=device-width" />
+		%sveltekit.head%
+	</head>
```
Comment (jeffmaury) [inline src/app.html:8]: Worth adding a title ?

File: src/lib/Appearance.spec.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  }
+  return rootElement as HTMLElement;
+}
+
+function getRootElementClassesValue(container: HTMLElement): string | undefined {
+  return getRootElement(container).classList.value;
+}
+
+test('check initial light theme', async () => {
+  const { baseElement } = render(Appearance, {});
```
Comment (jeffmaury) [inline src/lib/Appearance.spec.ts:35]:
```suggestion
  const { baseElement } = render(Appearance);
```

File: src/lib/ui/ExtensionByCategoryCard.svelte [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  // add a "v" in front of the version number if it's not there
+  return version.startsWith('v') ? version : `v${version}`;
+});
+
+function hideDetails(): void {
+  setCurrentExtension(extension);
+}
+</script>
+
+<button
```
Comment (jeffmaury) [inline src/lib/ui/ExtensionByCategoryCard.svelte:19]: Button componenrt ?

File: src/routes/page.svelte.spec.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import { render, screen } from '@testing-library/svelte';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import { catalogExtensions } from '$lib/extensions.svelte';
+
+import catalogOfExtensions from '../../static/api/extensions.json';
+import Page from '././+page.svelte';
```
Comment (jeffmaury) [inline src/routes/page.svelte.spec.ts:7]:
```suggestion
import Page from './+page.svelte';
```

**Review by jeffmaury — COMMENTED** [review]
Remarks:

- UI is quite rude, maybe worth adding a header ?
- Seems to be some components are duplicated, maybe follow up PR to get rid of it
- this PR holds a copy of the catalog so is it temporary
- When we list extension versions maybe worth having a link to install that specific version
- Validating workflow missing, follow up PR ?

File: src/lib/ui/ExtensionByCategoryCard.svelte (hunk 6 above)
Comment (benoitf) [reply src/lib/ui/ExtensionByCategoryCard.svelte:19]: so here it's using button to say that we have an active element with a onclick trigger (a card) but it doesn't look at all as a button

**Comment by benoitf** [comment]
> UI is quite rude, maybe worth adding a header ?

You'll will see it through podman desktop website
cf https://github.com/containers/podman-desktop-catalog/pull/68#issuecomment-2426172838

**Comment by benoitf** [comment]
> Seems to be some components are duplicated, maybe follow up PR to get rid of it

what are the components duplicated ?

**Comment by benoitf** [comment]
> this PR holds a copy of the catalog so is it temporary

it's not, it's how the files will be served (you won't need to create a PR to the gh-pages branch, it will be against this folder)

**Comment by benoitf** [comment]
> When we list extension versions maybe worth having a link to install that specific version

you don't have the ability to install a specific version

**Comment by slemeur** [comment]
This is looking great as a first step, I think we can move forward with this and create follow-up issues for improving the experience and the UI accordingly with the [mockups](https://github.com/containers/podman-desktop/issues/2363) on the points mentioned before.

**Comment by benoitf** [comment]
first batch: https://github.com/containers/podman-desktop-catalog/pull/69 (setup of the project)

I've included GH action

**Comment by benoitf** [comment]
replaced by https://github.com/containers/podman-desktop-catalog/pull/72

**Pull request closed without merging**